### ISSUE TITLE
Doesn't build when using webextension-toolbox-config.js

### ISSUE THREAD

**Issue by pbek** [webextension-toolbox/webextension-toolbox#499]
## Expected Behavior
- webextension-toolbox boilerplate project builds without errors while using `webextension-toolbox-config.js`

## Current Behavior
- Error message `webpack compiled with 82 errors and 10 warnings`

## Steps to Reproduce (for bugs)
1. create project with `yo web-extension` like suggested [here](https://github.com/webextension-toolbox/generator-web-extension#getting-started) (project builds at this point)
2. add `app/webextension-toolbox-config.js` with just one line: `const webpack = require('webpack')` like described [here](https://github.com/webextension-toolbox/webextension-toolbox#customizing-webpack-config)
3. execute `npm run build firefox`
4. see that it results in `webpack compiled with 82 errors and 10 warnings`

### Example error message

```
ERROR in ../node_modules/webpack/lib/webpack.js 8:13-28
Module not found: Error: Can't resolve 'util' in '/home/omega/Code/Test/my-web-extension/node_modules/webpack/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }
 @ ../node_modules/webpack/lib/index.js 95:30-50 98:9-29
 @ ./webextension-toolbox-config.js 1:16-34
```

## Context

I can't use vue.js without `webextension-toolbox-config.js`.

## Your Environment

| Tech                      | Version   |
| ------------------------- | --------- |
| webextension-toolbox      | 4.0.3     |
| node                      | v12.22.2  |
| OS                        | KDE Neon  |
| browser                   | Firefox 89.0.2    |

**Comment by onikienko** [comment]
> 2. add `app/webextension-toolbox-config.js` with just one line: const webpack = require('webpack') like described here

@pbek You have to add `webextension-toolbox-config.js` to the project root. Not into `app` folder

**Comment by pbek** [comment]
Originally (at version 3) I had it in the root directory, but the file gets ignored completely in the root directory in version 4 (see https://github.com/webextension-toolbox/webextension-toolbox/issues/309#issuecomment-865885103).

**Comment by pbek** [comment]
I just tried again with 4.0.3. The only directory where `webextension-toolbox-config.js` seems to get recognized is when it is in the `app` directory. Putting it into the project root (so one level up from `app`) work out for me.

**Comment by rthaut** [comment]
@pbek you need to name the file `webextension-toolbox.config.js` (note the period/dot between "toolbox" and "config"), as defined in the [main binary file/script here](https://github.com/webextension-toolbox/webextension-toolbox/blob/master/bin/webextension-toolbox#L12). With that name you can place it in the root directory and have it picked up automatically.

This appears to be an issue with the documentation; I will work on a PR to get it fixed.

**Comment by pbek** [comment]
Thank you for the hint. The file seems to be recognized now, but with a `const webpack = require('webpack')` in it (according to the documentation) you will now get `82 errors and 10 warnings`, like:

```
ERROR in ../node_modules/webpack/lib/util/fs.js 8:13-28
Module not found: Error: Can't resolve 'path' in '/home/omega/Test/web-extension/node_modules/webpack/lib/util'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }
 @ ../node_modules/webpack/lib/CleanPlugin.js 12:17-37
 @ ../node_modules/webpack/lib/index.js 141:9-33
 @ ./webextension-toolbox.config.js 1:16-34

ERROR in ../node_modules/webpack/lib/util/identifier.js 7:13-28
Module not found: Error: Can't resolve 'path' in '/home/omega/Test/web-extension/node_modules/webpack/lib/util'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }
 @ ../node_modules/webpack/lib/Compiler.js 33:30-58
 @ ../node_modules/webpack/lib/index.js 147:9-30
 @ ./webextension-toolbox.config.js 1:16-34

ERROR in ../node_modules/webpack/lib/webpack.js 8:13-28
Module not found: Error: Can't resolve 'util' in '/home/omega/Test/web-extension/node_modules/webpack/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }
 @ ../node_modules/webpack/lib/index.js 95:30-50 98:9-29
 @ ./webextension-toolbox.config.js 1:16-34
```

**Comment by onikienko** [comment]
@pbek 

I am not sure that it will help, but I do not require webpack `const webpack = require('webpack')` in my `webextension-toolbox.config.js`. And it works.

My file looks like:
```javaScript
module.exports = {
    webpack: (config) => {
            // my code here
        return config;
    }
};
```

**Comment by pbek** [comment]
@onikienko, that really helped. Thank you very much!
I needed to drop my:

```js
        config.plugins.push(new webpack.ProvidePlugin({
            Vue: 'vue',
            VueLoader: 'vue-loader'
        }));
```

But I could use:

```js
const VueLoaderPlugin = require('vue-loader/lib/plugin');
```

...at the start of `webextension-toolbox.config.js`.

Seems like https://github.com/webextension-toolbox/webextension-toolbox#customizing-webpack-config need even more changes.

**Comment by pbek** [comment]
I created a PR https://github.com/webextension-toolbox/webextension-toolbox/pull/515 for the documentation change.

**Comment by tm1000** [comment]
Closed from #515
